plugins/module_utils/network/junos/argspec/acls/acls.py

```diff
@@ -60,7 +60,10 @@ def __init__(self, **kwargs):
                                 "source": {
                                     "type": "dict",
                                     "options": {
-                                        "address": {"type": "str"},
+                                        "address": {
```

Comment:
As mentioned in #190 there's a diverse set of different rulesets out there in the real world:

```
from {
     source-address {
         192.168.1.0/32;
     }
}
```

```
from {
     source-address {
         192.168.1.0/32;
         192.168.2.0/32;
     }
}
```

I need to be able to support both cases but am a bit unsure how this can be achieved without making a potentially breaking change for users who already have the single address string in their group/hostvars.

Here we're bluntly changing the spec addresses to be a list. And in the facter/config I'm trying to handle both non-list and list cases, but that seems to be yielding curious behavior in the deletion integration test where the addresses are then appearing as a string that contains a dict with an extraneous "name". So most likely doing something wrong in the facter/config part - pointers appreciated.

Reply:
Ok sorted. Indeed there seems to be a name for an element regardless if it's defined in the argspec or not.